### ISSUE TITLE
Flatpak app fails to access repositories "Something went wrong"

### ISSUE THREAD

**Issue by DW8Reaper** [gitbutlerapp/gitbutler#3153]
I don't know if you support the Flatpak app but it was the default package installed from the PopShop on PopOS 22.04:
```terminal
➜  ~ flatpak list
Name                               Application ID                                   Version        Branch      Installation
...
GitButler Inc.                     com.gitbutler.gitbutler                          0.10.24        stable      user
```

Once installed you can start GitButler but it will not be able to access any repositories in your home folder which is does not have access to by default. The error message is not very helpful just "Something went wrong". 
![Screenshot from 2024-03-14 14-20-09](https://github.com/gitbutlerapp/gitbutler/assets/3433459/97e0aafb-cf5b-470d-8a7d-920a829dd70e)

This issue is to maybe improve the error message with a bit more detail or perhaps the Flatpak version should by default request access to a users home folder. 

I was able to work around this by custom configuration in flatseal for my repositories and also my ssh config for example:

![Screenshot from 2024-03-14 14-18-22](https://github.com/gitbutlerapp/gitbutler/assets/3433459/d619fb1b-1de5-44cc-a58d-407c287529b5)

**Comment by zanbaldwin** [comment]
To add to this, perhaps there should be an option in the Flatpak version of GitButler to use (or automatically try) `flatpak-spawn --host "${GIT_EXECUTABLE}"`.

I would much prefer this as the GitButler Flatpak should not have access to my SSH keys (kinda the point of Flatpak), but the `git` executable on my host does.

**Comment by honnip** [comment]
I used the flatpak app with various permissions modified, and yes it would be better to have the `filesystem=host` by default. 
Users can narrow the permissions if they want.

> `flatpak-spawn --host "${GIT_EXECUTABLE}"`

Yes, I think that's a better way to handle the authentication, and it would be nice to have some guide in the app.

**Comment by Qix-** [comment]
Hi there - if I provided a hook for `GIT_EXE` and allowed you to `export GIT_EXE='flatpak-spawn --host git'`, that would work, correct? Does that forward environment varables?

Also, we use some sidecar executables for setsid and askpass to forward passphrase prompts to the client that communicate via unix domain socket - will the host process have access to those? I've never personally use flatpak so I'm not sure what exactly happens in such an environment.

Thanks!

**Comment by DW8Reaper** [comment]
Ok I'm going to preface this with this is literally the first time I ever use flatpak-spawn but here is a test I did (on a side note you need to allow "socket=session-bus" for the flatpak app for flatpak-spawn to work I don't know how much of a security concern that is for peopel):

Run some flatpak and enter a terminal, I just used Brave since I had the flatpak on my machine 
```terminal
flatpak run --command=bash com.brave.Browser
```

Then in the flatpak terminal I ran this
```terminal
TEST_ME=some-test flatpak-spawn --env=TEST_ME2=ABC --host  env
```
and the result is `TEST_ME` is not set (so no it doesn't automatically export the ENV from the flatpak container) but the extra `TEST_ME2` I manually set worked like a champ. 

Which environment variables do you want it to have? It should to my understanding (and also in my test) have the environment vars from the host since that is where it is running

**Comment by ndom91** [comment]
Hey folks, we finally finished the Tauri v2 update and are working on a first-party flatpak! @honnip I reached out to you a few days ago via discord dm - we're still happy to collaborate here :grin: 

We've love to get some initial feedback of an early build. I believe `push`-ing to HTTPS remotes still isn't working due to it not being able to pop-up the auth dialog.

- Early build (based off of `master` as of 12.11.24): https://drive.google.com/file/d/1N-ZScmvj16jgZF05dHYOJzS84gfV0w6S/view?usp=sharing
- PR: https://github.com/gitbutlerapp/gitbutler/pull/5473
